### ISSUE TITLE
Add --diff-filter option to diff popup

### ISSUE THREAD

**Issue by dpapavas** [magit/magit#3661]
As the title says, I'd like to propose adding the `--diff-filter` option to the diff popup.  It allows filtering by file status, which can be useful in certain circumstances.  For instance, if, as part of the same commit, you've had to add several new files, and make changes to other existing ones, it can be useful to see only the one set or the other.  Similar arguments would apply for other categories, such as unmerged or renamed files, etc.

Right now, I've added the functionality with:

```lisp
  (magit-define-popup-option 'magit-diff-popup
                             ?F "Filter by file status" "--diff-filter=" #'read-from-minibuffer))
```

Let me know if you'd like me to try to create a patch and issue a pull request.

**Comment by tarsius** [comment]
From the issue template you saw when writing the above:

> Before you ask for a new argument to be added to some popup, see this [2] page.
> [2]: https://github.com/magit/magit/wiki/Additional-proposed-infix-arguments-and-suffix-commands

**Comment by dpapavas** [comment]
I'm not sure what you mean by the above.  You seem to imply that I ignored the suggestion in the template, but I did read the page you link to, which is how I found out that I can implement this change myself, something which I mentioned explicitly in my initial post.

The page did not mention that suggestions for the support of new options are not welcome, just that they might be rejected by the maintainer.  I suppose that's the case, and if it is, it's fine, but simply stating it would have been enough, and less confusing.
